### PR TITLE
Update Developer Docs for Wiki

### PR DESCRIPTION
**Pull request by HashWarlock**
Updated the developer section for the wiki with the following changes:

- Added tab navigation feature for each operating system compatible with developing on Phala Network with Ubuntu, Fedora, Windows WSL2 & MacOS (recommends users to use a VM running Ubuntu)
- Made minor typo changes to reflect the development steps correctly
- Moved the developer specifics images to their own folder within the static folder structure

### CODE REVIEW

**Comment by HashWarlock** [comment]
my-doks-site folder causing error, but should be resolved in PR #37

**Comment by HashWarlock** [comment]
Would like to understand if we will move the images to be referenced like they are in the current wiki by pointing to the repo here https://github.com/phala-wiki/phala-wiki.github.io/tree/3abcfeb01721f54a26265b20665766900150bdec

**Comment by hauni97** [comment]
I really like the tab navigation features.
* Removal of some content as in this commit https://github.com/Phala-Network/phala-wiki-next/pull/38/commits/90882464ee08ceef4ce3be2c2f9ebe7c0bb1c05e cannibalises this PR: https://github.com/Phala-Network/phala-wiki-next/pull/26 
  *  The image of the architecture should remain, as it would remove this in the newest version as shown below:

Screenshot 2021-12-03 at 16 24 20

* In commit https://github.com/Phala-Network/phala-wiki-next/pull/38/commits/09c9efb12b6b88bd7247619d3ae1a26e8b6464f0 you remove the script & GIF. It has the purpose of making the installation easier for new and inexperienced users that were merged here: https://github.com/Phala-Network/phala-wiki-next/pull/26 and should not be removed from my point of view.
   * I suggest improving the `auto install` script in this case and leaving or changing the GIF when the script has been updated. 
   *  The newly added content for Windows looks good.

**Comment by HashWarlock** [comment]
> I really like the tab navigation features.
> 
> * Removal of some content as in this commit [9088246](https://github.com/Phala-Network/phala-wiki-next/commit/90882464ee08ceef4ce3be2c2f9ebe7c0bb1c05e) cannibalises this PR: [general improvements #26](https://github.com/Phala-Network/phala-wiki-next/pull/26)
>   
>   * The image of the architecture should remain, as it would remove this in the newest version as shown below:
> 
> Screenshot 2021-12-03 at 16 24 20
> 
> * In commit [09c9efb](https://github.com/Phala-Network/phala-wiki-next/commit/09c9efb12b6b88bd7247619d3ae1a26e8b6464f0) you remove the script & GIF. It has the purpose of making the installation easier for new and inexperienced users that were merged here: [general improvements #26](https://github.com/Phala-Network/phala-wiki-next/pull/26) and should not be removed from my point of view.
>   
>   * I suggest improving the `auto install` script in this case and leaving or changing the GIF when the script has been updated.
>   * The newly added content for Windows looks good.

Thanks for catching this. I wasn't sure if this was outdated & required removal to reflect current wiki. I will go ahead and fix these issues

**Comment by HashWarlock** [comment]
@hauni97 updated the changes. Let me know if I missed anything.

**Comment by hauni97** [comment]
> @hauni97 updated the changes. Let me know if I missed anything.

Great improvements. Some missing details are:
* The images throughout the wiki have been simplified and are now loaded following this standardized format  `/images/docs/{wiki_topic}/{file_name}` please refer to this commit for more details: https://github.com/Phala-Network/phala-wiki-next/pull/35/commits/3fc91833f562948911e384573267c90edc410225 else this PR will conflict
* The installation of `clang`, `Rust`& `Intel SGX drivers` are already handled in the installation script; hence it leads to a duplication in the material when selecting the`Ubuntu` tab as seen here:
image

I suggest auditing & pulling https://github.com/Phala-Network/phala-wiki-next/pull/38 later (placing on hold) and adding the tabs instead of collapsing sections in the `developer` wiki guide + adding your installation guide for `Fedora` when https://github.com/Phala-Network/phala-wiki-next/pull/35 https://github.com/Phala-Network/phala-wiki-next/pull/36 https://github.com/Phala-Network/phala-wiki-next/pull/37 has been merged to master.

File: content/en-us/docs/developer/run-a-local-development-network.md [hunk 1]
```diff
@@ -32,91 +32,209 @@ It's required to have at least 4 cores and 8GB RAM to build the project includin
...
+  <div class="tab-pane" id="messages" role="tabpanel" aria-labelledby="messages-tab">
+  <br>
 
-  ```bash
-  sudo apt update
-  sudo apt install -y build-essential ocaml ocamlbuild automake autoconf libtool wget python libssl-dev git cmake perl pkg-config curl llvm-10 clang-10 libclang-10-dev
-  ```
+> Instructions for (but not limited to) a relatively clean Windows11/WSL2 environment (Ubuntu 20.04LTS, and git correctly installed) are preceded by the mention "For WSL2"
+>> Install WSL2 (Windows 10 version 2004 and higher (Build 19041 and higher) or Windows 11)
+>>More infos at: https://docs.microsoft.com/en-us/windows/wsl/install
```
Comment (hauni97) [inline content/en-us/docs/developer/run-a-local-development-network.md:182]: I suggest linking it here instead of the using the plain URL.

**Review by hauni97 — CHANGES_REQUESTED** [review]

**Comment by hauni97** [comment]
@HashWarlock maybe fetch the upstream once the other PR's are merged, this may save time.

**Comment by HashWarlock** [comment]
Yeah, I can rebase when PR's are merged then I'll make changes with the tabs.

File: content/en-us/docs/developer/run-a-local-development-network.md [hunk 2]
```diff
@@ -32,91 +32,209 @@ It's required to have at least 4 cores and 8GB RAM to build the project includin
...
+sudo npm install -g yarn
+```
+  </div>
+  <div class="tab-pane" id="settings" role="tabpanel" aria-labelledby="settings-tab">
+  <br>
 
-### Mac
+> MacOS users can opt to use a Virtual Machine running Ubuntu and follow the Ubuntu configuration steps.
 
-> Currently, Mac is not supported. However, if you wish to run a simulated Phala environment for this tutorial, you may create a Virtual Machine (VM). Furthermore, MacOS is a UNIX-based Operating System; you may easily use SSH to connect to your VM without additional tools. A free resource is Amazon Web Services <a href="https://aws.amazon.com/getting-started/launch-a-virtual-machine-B-0/" target="_blank">AWS</a>.
```
[the remark this answers is not in the document]
Comment (HashWarlock) [reply content/en-us/docs/developer/run-a-local-development-network.md:108]: Adding this to the MacOS Tab in the docs to resolve this.

**Comment by HashWarlock** [comment]
> 

On this part, I differentiate them by stating the prior is installation is via script & the other one via manual installation. That was how it looked in the previous commit. Would we want to leave the manual installation process in the docs or would it be better to use collapsible section for that?

File: content/en-us/docs/developer/run-a-local-development-network.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (HashWarlock) [reply content/en-us/docs/developer/run-a-local-development-network.md:108]: How does this look for the MacOS tab?
image

File: content/en-us/docs/developer/run-a-local-development-network.md (hunk 1 above)
Comment (HashWarlock) [reply content/en-us/docs/developer/run-a-local-development-network.md:182]: Updated to reflect a link on `Install WSL2`
image

**Comment by HashWarlock** [comment]
Waiting on PR #35 to be merged to pull in major updates into my branch

**Pull request merged**